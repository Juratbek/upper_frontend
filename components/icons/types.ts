export interface IIconProps {
  color?: string;
  variant?: 'outlined' | 'fulfilled';
  width?: number;
  height?: number;
  opacity?: number;
}
