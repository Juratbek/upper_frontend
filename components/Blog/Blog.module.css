.name {
    margin-right: 2rem;
}

.bio {
    font-size: 1.1rem;
}

.avatar {
    margin-right: .6rem;
}