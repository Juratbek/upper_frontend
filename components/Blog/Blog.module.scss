.name {
  margin-right: 2rem;
}

.bio {
  font-size: 1.1rem;
}

.avatar {
  margin-right: 0.6rem;
}

.social-media-links {
  position: absolute;
  left: 0;
  bottom: -0.1rem;
  transform: translateY(100%);
  display: flex;

  & .icon {
    margin-right: 0.2rem;
    $container-size: 1.7rem;
    min-width: $container-size;
    height: $container-size;
    border-radius: 100rem;
    display: grid;
    place-items: center;

    &:hover {
      background-color: var(--lightGray);
    }
   
    & > svg {
      $size: 1.0rem;
      width: $size;
      height: $size;
    }
  }
}
