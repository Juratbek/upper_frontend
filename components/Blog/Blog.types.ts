import { TAvatarSize } from 'components/lib';
import { IBlogMedium, ILink } from 'types';

export interface IBlogProps extends IBlogMedium {
  className?: string;
  avatarSize?: TAvatarSize;
  isLink?: boolean;
  links?: ILink[];
}
