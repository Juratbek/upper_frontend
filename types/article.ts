import { OutputData } from '@editorjs/editorjs';

import { IBlog } from './blog';
import { ILabel } from './label';

export interface IArticleResult {
  id: number;
  title: string;
  content: string;
  imgUrl?: string;
  author?: IBlog;
  labels: ILabel[];
  publishedDate: Date;
  updatedDate?: Date;
  viewCount?: number;
  likes?: number;
}

export interface IArticle {
  id: number;
  title: string;
  content: OutputData;
  imgUrl?: string;
  author?: IBlog;
  labels: ILabel[];
  publishedDate: string;
  updatedDate?: string;
  viewCount?: number;
  likes?: number;
  prevArticleId?: number;
  nextArticleId?: number;
}
