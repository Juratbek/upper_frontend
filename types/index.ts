export * from './common';
export * from './article';
export * from './blog';
export * from './label';
export * from './notification';
export * from './auth';
