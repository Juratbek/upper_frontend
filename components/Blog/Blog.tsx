import { Avatar } from 'components/Avatar/Avatar';
import Link from 'next/link';
import { FC, useCallback } from 'react';
import { getClassName } from 'utils';

import classes from './Blog.module.css';
import { IBlogProps } from './Blog.types';

export const Blog: FC<IBlogProps> = ({ imgUrl, name, bio, avaratSize = 'large', ...props }) => {
  const { followersCount, className, isLink, id, articlesCount } = props;
  const rootClassName = getClassName(classes.blog, className);

  const getBlog = useCallback(
    (className?: string) => (
      <div className={`d-flex align-items-center mb-1 ${className}`}>
        <Avatar imgUrl={imgUrl} size={avaratSize} className={classes.avatar} />
        <div>
          <h2 className='m-0'>{name}</h2>
          <p className='m-0'>
            {followersCount && <span>{followersCount} kuzatuvchilar</span>}{' '}
            {articlesCount && <span>{articlesCount} maqolalar</span>}
          </p>
        </div>
      </div>
    ),
    [],
  );

  return (
    <div className={rootClassName}>
      {isLink ? <Link href={`/blogs/${id}`}>{getBlog('pointer')}</Link> : getBlog()}
      <div>{bio && <p className={`${classes.bio} m-0`}>{bio}</p>}</div>
    </div>
  );
};
