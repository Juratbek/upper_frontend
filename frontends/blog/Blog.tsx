import { Blog, Button, Head, Modal, TabBody, TabsHeader } from 'components';
import { useDevice } from 'hooks';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { FC, useEffect, useState } from 'react';
import { useFollowBlogMutation, useUnfollowBlogMutation } from 'store/apis';
import { addAmazonUri, convertBlogToHeadProp, get } from 'utils';
import { BLOG_TAB_MENUS, BLOG_TABS, ICONS } from 'variables';

import styles from './Blog.module.scss';
import { IBlogPageProps } from './Blog.types';

const HeartIcon = ICONS.heart;

export const BlogPage: FC<IBlogPageProps> = ({ blog, error, fullUrl }: IBlogPageProps) => {
  const [isUnfollowModalOpen, setIsUnfollowModalOpen] = useState<boolean>(false);
  const [isFollowed, setIsFollowed] = useState<boolean>(blog?.isFollowed || false);
  const { isMobile } = useDevice();
  const [followBlog, followBlogRes] = useFollowBlogMutation();
  const [unfollowBlog, unfollowBlogRes] = useUnfollowBlogMutation();
  const {
    query: { id },
  } = useRouter();

  const follow = (): void => {
    id && followBlog(+id).then(() => setIsFollowed(true));
  };

  const unfollow = (): void => {
    id &&
      unfollowBlog(+id).then(() => {
        setIsFollowed(false);
        setIsUnfollowModalOpen(false);
      });
  };

  const toggleUnfollowModal = (): void => {
    setIsUnfollowModalOpen((prev) => !prev);
  };

  useEffect(() => {
    if (blog?.isFollowed !== undefined) {
      setIsFollowed(blog.isFollowed);
    }
  }, [id]);

  if (!blog) return <h3>{get(error, 'data.message')}</h3>;

  return (
    <div className='container'>
      <Head {...convertBlogToHeadProp(blog)} url={fullUrl} />
      <Modal
        size='small'
        isOpen={isUnfollowModalOpen}
        close={toggleUnfollowModal}
        bodyClassName='text-center'
      >
        <strong>{blog.name}</strong> obunasini bekor qilmoqchimisiz
        <div className={['mt-2', styles.buttonsContainer].join(' ')}>
          <Button onClick={toggleUnfollowModal} color='outline-dark' className='me-2'>
            Yopish
          </Button>
          <Button onClick={unfollow} loading={unfollowBlogRes.isLoading} color='outline-dark'>
            Obunani bekor qilish
          </Button>
        </div>
      </Modal>

      <div
        className={`d-flex align-items-center justify-content-between p-2 flex-wrap ${styles.blogContainer}`}
      >
        {blog && (
          <>
            <Blog {...addAmazonUri(blog)} avaratSize='extra-large' className='mb-2 flex-1' />
            {!blog.isCurrentBlog && (
              <>
                {isFollowed ? (
                  <Button color='outline-dark' onClick={toggleUnfollowModal}>
                    Obuna bo`lingan
                  </Button>
                ) : (
                  <Button onClick={follow} loading={followBlogRes.isLoading}>
                    Obuna bo`lish
                  </Button>
                )}
                {Boolean(blog.cardNumber) && (
                  <Link href={`/blogs/${id}/support`}>
                    <a className='link d-flex mt-xs-2'>
                      {isMobile ? (
                        <Button className='w-100' color='outline-dark'>
                          <span className='sponsor-icon'>
                            <HeartIcon />
                          </span>
                          Blog faoliyatiga o&apos;z hissangizni qo&apos;shing
                        </Button>
                      ) : (
                        <>
                          <span className='sponsor-icon'>
                            <HeartIcon />
                          </span>
                          <h4 className='m-0'>
                            Blog faoliyatiga o&apos;z hissangizni qo&apos;shing
                          </h4>
                        </>
                      )}
                    </a>
                  </Link>
                )}
              </>
            )}
          </>
        )}
      </div>
      <TabsHeader tabs={BLOG_TAB_MENUS} />
      <TabBody tabs={BLOG_TABS} />
    </div>
  );
};
