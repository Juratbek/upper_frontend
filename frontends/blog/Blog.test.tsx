import { render } from '@testing-library/react';
import { QueryWrapper, ReduxWrapper } from 'tests';

import { BlogPage } from './Blog';
import { IBlogPageProps } from './Blog.types';

const mockProps: IBlogPageProps = {
  fullUrl: 'https://upper.uz/web/blogs/12',
  blog: {
    createdDate: '12.10.23',
    id: 12,
    imgUrl: 'https://image.upper.uz/123',
    name: 'Samandar',
    links: [],
    labels: [],
  },
  error: null,
};

const Page = (props: IBlogPageProps) => (
  <ReduxWrapper>
    <QueryWrapper>
      <BlogPage {...props} />
    </QueryWrapper>
  </ReduxWrapper>
);

describe('BlogPage', () => {
  it('snapshot', () => {
    render(Page(mockProps));
    expect(document.body).toMatchSnapshot();
  });

  it('error UI snapshot', () => {
    mockProps.blog = null;
    mockProps.error = { status: 404, data: { code: 404, message: 'Blog topilmadi' } };
    render(Page(mockProps));
    expect(document.body).toMatchSnapshot();
  });
});
