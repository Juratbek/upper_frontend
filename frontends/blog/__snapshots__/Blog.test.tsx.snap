// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlogPage > error UI snapshot 1`] = `
<body>
  <div>
    <h3
      class="text-center flex-1"
    >
      Blog topilmadi
    </h3>
  </div>
</body>
`;

exports[`BlogPage > snapshot 1`] = `
<body>
  <div>
    <div
      class="flex-1"
    >
      <header
        class="header container"
      >
        <div
          class="search-container hidden"
        >
          <span>
            <svg
              fill="none"
              height="24"
              role="icon"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M15 6L9 12L15 18"
                stroke="black"
                stroke-width="1.6"
              />
            </svg>
          </span>
          <div
            class="position-relative w-100"
          >
            <input
              class="search-input input"
              style="border-color: rgba(0, 0, 0, 0.25);"
            />
          </div>
        </div>
        <a
          class="logo link"
          href="/web"
        >
          <svg
            data-testid="logo-svg"
            fill="none"
            role="img"
            viewBox="0 0 188 80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              fill="white"
              id="path-1-inside-1_804_345"
            >
              <path
                d="M0 0H80V80H0V0Z"
              />
            </mask>
            <path
              d="M0 0V-5H-5V0H0ZM0 80H-5V85H0V80ZM0 5H80V-5H0V5ZM80 75H0V85H80V75ZM5 80V0H-5V80H5Z"
              fill="#54A9EB"
              mask="url(#path-1-inside-1_804_345)"
            />
            <line
              stroke="#54A9EB"
              stroke-width="5"
              x1="77.5"
              x2="77.5"
              y1="5"
              y2="17"
            />
            <line
              stroke="#54A9EB"
              stroke-width="5"
              x1="77.5"
              x2="77.5"
              y1="63"
              y2="75"
            />
            <path
              d="M50.8324 21.9091H55.3239V46.4673C55.3239 49.0028 54.7262 51.2667 53.5309 53.2589C52.3477 55.239 50.6754 56.8026 48.5142 57.9496C46.353 59.0845 43.8175 59.652 40.9077 59.652C37.9979 59.652 35.4624 59.0845 33.3011 57.9496C31.1399 56.8026 29.4616 55.239 28.2663 53.2589C27.0831 51.2667 26.4915 49.0028 26.4915 46.4673V21.9091H30.983V46.1051C30.983 47.9162 31.3814 49.5281 32.1783 50.9407C32.9751 52.3413 34.1101 53.446 35.5831 54.255C37.0682 55.0518 38.843 55.4503 40.9077 55.4503C42.9723 55.4503 44.7472 55.0518 46.2322 54.255C47.7173 53.446 48.8523 52.3413 49.6371 50.9407C50.434 49.5281 50.8324 47.9162 50.8324 46.1051V21.9091Z"
              fill="#54A9EB"
            />
            <path
              d="M64.2932 59V21.9091H76.8259C79.7357 21.9091 82.1143 22.4343 83.9616 23.4847C85.8209 24.5231 87.1974 25.9297 88.0908 27.7045C88.9843 29.4794 89.431 31.4595 89.431 33.6449C89.431 35.8303 88.9843 37.8164 88.0908 39.6033C87.2094 41.3903 85.8451 42.815 83.9978 43.8775C82.1505 44.9279 79.784 45.4531 76.8983 45.4531H67.9154V41.4688H76.7535C78.7457 41.4688 80.3454 41.1246 81.5528 40.4364C82.7602 39.7482 83.6356 38.8185 84.1789 37.6474C84.7343 36.4641 85.012 35.13 85.012 33.6449C85.012 32.1598 84.7343 30.8317 84.1789 29.6605C83.6356 28.4893 82.7542 27.5717 81.5347 26.9077C80.3153 26.2315 78.6974 25.8935 76.681 25.8935H68.7847V59H64.2932ZM96.6663 59V21.9091H109.199C112.109 21.9091 114.487 22.4343 116.335 23.4847C118.194 24.5231 119.57 25.9297 120.464 27.7045C121.357 29.4794 121.804 31.4595 121.804 33.6449C121.804 35.8303 121.357 37.8164 120.464 39.6033C119.582 41.3903 118.218 42.815 116.371 43.8775C114.524 44.9279 112.157 45.4531 109.271 45.4531H100.288V41.4688H109.127C111.119 41.4688 112.718 41.1246 113.926 40.4364C115.133 39.7482 116.009 38.8185 116.552 37.6474C117.107 36.4641 117.385 35.13 117.385 33.6449C117.385 32.1598 117.107 30.8317 116.552 29.6605C116.009 28.4893 115.127 27.5717 113.908 26.9077C112.688 26.2315 111.07 25.8935 109.054 25.8935H101.158V59H96.6663ZM129.039 59V21.9091H151.424V25.8935H133.531V38.4261H150.265V42.4105H133.531V55.0156H151.714V59H129.039ZM159.52 59V21.9091H172.052C174.95 21.9091 177.329 22.4041 179.188 23.3942C181.047 24.3722 182.424 25.7184 183.317 27.4329C184.211 29.1474 184.658 31.0973 184.658 33.2827C184.658 35.468 184.211 37.4059 183.317 39.0962C182.424 40.7866 181.054 42.1147 179.206 43.0806C177.359 44.0344 174.998 44.5114 172.125 44.5114H161.983V40.4545H171.98C173.96 40.4545 175.554 40.1648 176.761 39.5852C177.981 39.0057 178.862 38.1847 179.405 37.1222C179.961 36.0476 180.239 34.7678 180.239 33.2827C180.239 31.7976 179.961 30.4996 179.405 29.3888C178.85 28.2781 177.963 27.4208 176.743 26.8171C175.524 26.2013 173.912 25.8935 171.908 25.8935H164.011V59H159.52ZM176.979 42.3381L186.106 59H180.891L171.908 42.3381H176.979Z"
              fill="black"
            />
          </svg>
        </a>
        <span
          class="search-icon"
        >
          <svg
            fill="none"
            height="24"
            role="icon"
            viewBox="0 0 20 20"
            width="24"
          >
            <path
              d="M17.4344 18.3334C17.204 18.3334 16.9736 18.2437 16.8072 18.0772L14.4262 15.6944C14.0806 15.3485 14.0806 14.7849 14.4262 14.4262C14.7718 14.0803 15.3351 14.0803 15.6935 14.4262L18.0744 16.8089C18.4201 17.1548 18.4201 17.7185 18.0744 18.0772C17.8952 18.2437 17.6648 18.3334 17.4344 18.3334Z"
              fill="black"
              stroke="#F5F5F5"
              stroke-width="0.6"
            />
            <path
              d="M16.8003 9.58341C16.8003 13.5691 13.5693 16.8001 9.58366 16.8001C5.598 16.8001 2.36699 13.5691 2.36699 9.58341C2.36699 5.59776 5.598 2.36675 9.58366 2.36675C13.5693 2.36675 16.8003 5.59776 16.8003 9.58341Z"
              stroke="black"
              stroke-width="1.4"
            />
          </svg>
        </span>
      </header>
      <div
        class="root"
      >
        <nav
          class="navigation"
          style="top: 2rem;"
        >
          <div
            class="root"
          >
            <div
              class="content"
            >
              <a
                class="navigation-button link"
                href="/web"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="none"
                    height="24"
                    role="icon"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M9 17V21.9997H5C3.89543 21.9997 3 21.1043 3 19.9997V10.0332C3 9.73801 3.13046 9.45786 3.35643 9.26786L10.7129 3.08228C11.4569 2.45664 12.5431 2.45665 13.2871 3.08228L20.6436 9.26786C20.8695 9.45786 21 9.73801 21 10.0332V20C21 21.1046 20.1046 22 19 22H15V17C15 13 9 13 9 17Z"
                      fill="black"
                    />
                  </svg>
                </span>
                <span
                  class="label"
                >
                  Asosiy
                </span>
              </a>
              <a
                class="navigation-button link"
                href="/web/user/articles/draft"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="none"
                    height="24"
                    role="icon"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M12 4H5.77778C5.30628 4 4.8541 4.1873 4.5207 4.5207C4.1873 4.8541 4 5.30628 4 5.77778V18.2222C4 18.6937 4.1873 19.1459 4.5207 19.4793C4.8541 19.8127 5.30628 20 5.77778 20H18.2222C18.6937 20 19.1459 19.8127 19.4793 19.4793C19.8127 19.1459 20 18.6937 20 18.2222V12"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.6"
                    />
                    <path
                      d="M17.9227 3.52798C18.2607 3.18992 18.7193 3 19.1973 3C19.6754 3 20.134 3.18992 20.472 3.52798C20.8101 3.86605 21 4.32456 21 4.80265C21 5.28075 20.8101 5.73926 20.472 6.07732L12.3991 14.1502L9 15L9.84978 11.6009L17.9227 3.52798Z"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.6"
                    />
                  </svg>
                </span>
                <span
                  class="label"
                >
                  Maqolalaringiz
                </span>
              </a>
              <button
                class="add-btn"
              >
                <span
                  class="circle"
                >
                  <div
                    color="black"
                    height="44"
                    role="icon"
                    width="44"
                  >
                    +
                  </div>
                </span>
              </button>
              <a
                class="navigation-button link"
                href="/web/notifications"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="none"
                    height="24"
                    role="icon"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <mask
                      fill="white"
                      id="path-2-inside-1_630_4298"
                    >
                      <path
                        d="M4.5 10.5C4.5 6.35786 7.85786 3 12 3C16.1421 3 19.5 6.35786 19.5 10.5V17C19.5 17.5523 19.0523 18 18.5 18H5.5C4.94772 18 4.5 17.5523 4.5 17V10.5Z"
                      />
                    </mask>
                    <path
                      d="M4.5 10.5C4.5 6.35786 7.85786 3 12 3C16.1421 3 19.5 6.35786 19.5 10.5V17C19.5 17.5523 19.0523 18 18.5 18H5.5C4.94772 18 4.5 17.5523 4.5 17V10.5Z"
                      mask="url(#path-2-inside-1_630_4298)"
                      stroke="black"
                      stroke-width="2.8"
                    />
                    <path
                      d="M16 17C16 17.5253 15.8965 18.0454 15.6955 18.5307C15.4945 19.016 15.1999 19.457 14.8284 19.8284C14.457 20.1999 14.016 20.4945 13.5307 20.6955C13.0454 20.8965 12.5253 21 12 21C11.4747 21 10.9546 20.8965 10.4693 20.6955C9.98396 20.4945 9.54301 20.1999 9.17157 19.8284C8.80014 19.457 8.5055 19.016 8.30448 18.5307C8.10346 18.0454 8 17.5253 8 17L12 17H16Z"
                      fill="black"
                    />
                    <path
                      d="M7 18H17V19H7V18Z"
                      fill="white"
                    />
                  </svg>
                </span>
                <span
                  class="label"
                >
                  Xabarlar
                </span>
              </a>
              <a
                class="navigation-button link"
                href="/web/profile"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="none"
                    height="24"
                    role="icon"
                    viewBox="0 0 29 29"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M19.4208 4.86294C18.2534 3.67557 16.606 3.02087 14.6836 3.02087C12.7613 3.02087 11.1139 3.67557 9.94655 4.86294C8.78013 6.04939 8.13981 7.71997 8.13981 9.66671C8.13981 12.2876 9.28093 13.9768 10.6932 14.9909C10.9677 15.1878 11.2513 15.3588 11.5375 15.5062C9.95921 15.9399 8.67769 16.6959 7.63431 17.6337C5.98448 19.1165 4.96193 21.0242 4.27301 22.7324C4.14822 23.0419 4.2979 23.3939 4.60737 23.5187C4.91682 23.6435 5.26885 23.4938 5.39364 23.1843C6.04937 21.5584 6.98588 19.8411 8.442 18.5324C9.88351 17.2369 11.8659 16.3125 14.6836 16.3125C17.3178 16.3125 19.2195 17.1203 20.631 18.2796C22.0544 19.4485 23.013 21.0028 23.6998 22.5393C24.158 23.5645 23.3539 24.7709 22.0915 24.7709H8.15001C7.81634 24.7709 7.54584 25.0414 7.54584 25.375C7.54584 25.7087 7.81634 25.9792 8.15001 25.9792H22.0915C24.0878 25.9792 25.6745 23.9961 24.803 22.0463C24.0738 20.4151 23.0192 18.6771 21.398 17.3458C20.4356 16.5555 19.2853 15.9183 17.9078 15.528C18.6445 15.1206 19.2501 14.58 19.732 13.9673C20.7747 12.6416 21.2275 10.9961 21.2275 9.66671C21.2275 7.71997 20.5872 6.04939 19.4208 4.86294ZM18.5591 5.71006C19.4715 6.63819 20.0191 7.98844 20.0191 9.66671C20.0191 10.754 19.6403 12.1293 18.7823 13.2203C17.9407 14.2904 16.6275 15.1042 14.6836 15.1042C13.8975 15.1042 12.5411 14.83 11.398 14.0092C10.2859 13.2107 9.34815 11.8792 9.34815 9.66671C9.34815 7.98844 9.89575 6.63819 10.8082 5.71006C11.7198 4.78284 13.0421 4.22921 14.6836 4.22921C16.3251 4.22921 17.6475 4.78284 18.5591 5.71006Z"
                      fill="black"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="label"
                >
                  Profil
                </span>
              </a>
            </div>
          </div>
        </nav>
        <main
          class="main container"
        >
          <div
            class="modal"
            data-testid="modal-root"
          >
            <button
              class="root modal-bg"
            />
            <div
              class="modal-dialog modal-dialog--undefined"
            >
              <div
                class="modal-body"
              >
                <p
                  class="text"
                >
                  Obunani bekor qilmoqchimisiz?
                </p>
              </div>
              <div
                class="d-flex"
              >
                <button
                  class="root unsubscribe-btn"
                >
                  Bekor qilish
                </button>
                <button
                  class="root no-btn"
                >
                  yo'q
                </button>
              </div>
            </div>
          </div>
          <div
            class="d-flex align-items-center justify-content-between p-2 flex-wrap blogContainer"
          >
            <div
              class="blog mb-2 flex-1"
            >
              <div
                class="d-flex align-items-center mb-1 undefined"
              >
                <div
                  class="avatar avatar--extra-large avatar"
                  style="border-color: rgba(245, 245, 245, 0.1);"
                >
                  <img
                    alt="UPPER"
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    sizes="100vw"
                    src="http://localhost:3000/_next/image?url=https%3A%2F%2Fimage.upper.uz%2F123&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fimage.upper.uz%2F123&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F123&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F123&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F123&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F123&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F123&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F123&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F123&w=3840&q=75 3840w"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
                  />
                </div>
                <div
                  class="position-relative flex-1"
                >
                  <h2
                    class="m-0"
                  >
                    Samandar
                  </h2>
                </div>
              </div>
              <div />
            </div>
          </div>
          <div>
            Yuklanmoqda...
          </div>
        </main>
        <aside
          class="sidebar"
          style="top: 2rem;"
        >
          <div
            class="add-container"
          >
            <h4
              class="sale"
            >
              Chegirma
            </h4>
            <h3
              class="percent"
            >
              15%
            </h3>
            <p
              class="text"
            >
              UPPER promokodi bilan Frontint.uz kurslariga chegirma oling
            </p>
          </div>
          <a
            href="https://frontint.uz/"
            rel="noreferrer"
            target="_blank"
          >
            <button
              class="button mt-1 w-100 btn"
              style="opacity: 1;"
            >
              Chegirma olish
            </button>
          </a>
        </aside>
      </div>
    </div>
  </div>
</body>
`;
